### PR TITLE
feat: Collecting community messages count metrics

### PR DESCRIPTION
**Pull request by MishkaRogachev**
For https://github.com/status-im/status-desktop/issues/11152

Implement both strategies for fetching community messaging metrics:

- [x] `CommunityMetricsRequestMessagesTimestamps` to collect messages timestamps by intervals for all community chats
- [x] `CommunityMetricsRequestMessagesCount` to collect messages count by intervals for all community chats

### CODE REVIEW

File: protocol/persistence_metrics.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	var timestamps []uint64
+	for _, chatID := range chatIDs {
```
Comment (MishkaRogachev) [inline protocol/persistence_metrics.go]: Here I get all the chat id's for the community, but I can probably rely on the fact that the channel id is communityId+uuid, and so I can just select messages with the chat id starting with the community id

**Review by alexjba — APPROVED** [review]
LGTM! 👍

**Review by jrainville — APPROVED** [review]
Nice job. This is very clean

File: protocol/messenger_community_metrics.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	return community.ChatIDs(), nil
+}
+
+func (m *Messenger) collectCommunityMessagesTimestamps(request *requests.CommunityMetricsRequest) (*CommunityMetricsResponse, error) {
+	chatIDs, err := m.getChatIdsForCommunity(request.CommunityID)
+	if err != nil {
+		return nil, err
+	}
+
+	intervals := []MetricsIntervalResponse{}
```
Comment (ilmotta) [inline protocol/messenger_community_metrics.go]: Whenever possible, it's a good idea to initialize slices with a max size (2nd arg to `make`). In this case the length of `request.Intervals`

File: protocol/persistence_metrics.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	query := fmt.Sprintf(selectTimestampsQuery, querySeveralChats(chatIDs))
+
+	rows, err := db.db.Query(query, startTimestamp, endTimestamp)
+	if err != nil {
+		return []uint64{}, err
+	}
+	defer rows.Close()
+
+	var timestamps []uint64
+	for rows.Next() {
```
Comment (ilmotta) [inline protocol/persistence_metrics.go:34]: The section `Fetching Data from the Database` is quite useful, as is the whole page actually http://go-database-sql.org/retrieving.html. It says we should ideally always check for errors after looping with `rows.Err()`. It's not enough to just `defer rows.Close()`.

File: protocol/requests/community_metrics_request.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	EndTimestamp   uint64 `json:"endTimestamp"`
+}
+
+type CommunityMetricsRequest struct {
+	CommunityID types.HexBytes              `json:"communityId"`
+	Type        CommunityMetricsRequestType `json:"type"`
+	Intervals   []MetricsIntervalRequest    `json:"intervals"`
+}
+
+func (r *CommunityMetricsRequest) Validate() error {
```
Comment (ilmotta) [inline protocol/requests/community_metrics_request.go:32]: We could have some tests for the validations, but it's fine without since it's fairly simple.

**Review by ilmotta — APPROVED** [review]
Very well done @MishkaRogachev! 🚀

File: protocol/messenger_community_metrics.go (hunk 2 above)
Comment (MishkaRogachev) [reply protocol/messenger_community_metrics.go]: fixed

File: protocol/persistence_metrics.go (hunk 3 above)
Comment (MishkaRogachev) [reply protocol/persistence_metrics.go:34]: fixed

File: protocol/requests/community_metrics_request.go (hunk 4 above)
Comment (MishkaRogachev) [reply protocol/requests/community_metrics_request.go:32]: added small test case checking invalid request intervals

**Pull request merged**